skimage/io/util.py

```diff
@@ -28,6 +28,8 @@ def file_or_url_context(resource_name):
                 f.write(u.read())
             # f must be closed before yielding
             yield f.name
+        except OSError:
+            raise ValueError('unreachable URL: {}'.format(resource_name))
```

Comment:
what is the original error? could you reraise that one instead?

Reply:
As mentioned in the original Issue (#3785), an `OSError: [Errno 22] Invalid argument` is raised followed by a `UnboundLocalError: local variable 'f' referenced before assignment`.
This error was not managed at all.